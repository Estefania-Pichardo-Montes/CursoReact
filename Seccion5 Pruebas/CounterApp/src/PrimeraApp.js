//Los componentes se escriben con el upper camelcase
//hay dos tipos de componentes, basados por clases o basados por funciones

//Se llaman funcional components
import React, { Fragment } from 'react';
import  PropTypes from 'prop-types';

//JavaScript solo puede regresar un objeto a la vez
//Tenemos que retornar los elementos HTML en un elemento que los contenga a todos
//Puedes utilizar un div para englobarlos o un fragment
//Si no le ponemos contenido a la etiqueta que engloba por default lo toma como un fragment

const PrimeraApp=({saludo, subtitulo})=>{

    //Los booleanos no los imprimira en pantalla

    return (
        <>
            <h1>{saludo}</h1>
            {/* Las llaves es para indicar lo que va a retornar de alguna cosntante */}
            {/* Para imprimir un objeto podemos utilizar el metodo Json.stringify */}
            {/* <pre>{JSON.stringify(saludo,null,3)}</pre> */}
            <p>{subtitulo}</p>
        </>

    );

}

PrimeraApp.propTypes={
    saludo:PropTypes.string.isRequired
}

PrimeraApp.defaultProps={
    subtitulo:'Soy un subtitulo'
}
export default PrimeraApp;

