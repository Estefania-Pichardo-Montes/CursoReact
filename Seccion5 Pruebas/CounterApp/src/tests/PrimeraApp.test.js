import React from 'react';
import {render} from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';
import setupTest from '../setupTest';
import '@testing-library/jest-dom'; 

describe('Pruebas en <PrimeraApp/>', () => {


    // test('debe Mostrar el mensaje "Hola soy Octavia"', () => {
    //     const saludo='Hola, soy Octavia';
    //     const {getByText}=render(<PrimeraApp saludo={saludo}/>)
    //     expect( getByText(saludo) ).toBeInTheDocument;
    // })
    
    test('debe de mostrar <primeraApp/> correctamente ', () => {
        //Shallow es muy parecido al render pero con mas opciones
        //Podemos simular clicks y tabajar con el documento

        const saludo='Hola, soy Octavia'
        const wrapper=shallow(<PrimeraApp saludo={saludo}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar el subtitulo mandado por props ', () => {
        
        const saludo='Hola, soy Octavia'
        const subTitulo='Soy un subtitulo'
        const wrapper=shallow(

            <PrimeraApp
             saludo={saludo}
             subtitulo={subTitulo}
             />
        );

        const textoParrafo=wrapper.find('p').text();
        console.log(textoParrafo);
        expect(textoParrafo).toBe(subTitulo);
    })
    
    
})
