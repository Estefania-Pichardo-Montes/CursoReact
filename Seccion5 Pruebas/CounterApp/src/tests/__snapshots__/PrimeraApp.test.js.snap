// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> debe de mostrar <primeraApp/> correctamente  1`] = `
<Fragment>
  <h1>
    Hola, soy Octavia
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
